src/java/org/apache/cassandra/db/streaming/CassandraStreamReceiver.java

```diff
@@ -242,6 +246,19 @@ public void finished()
             }
             else
             {
+                if (receivedEntireSSTable)
+                {
+                    try
+                    {
+                        // SSTable-attached indexes may have been streamed completely, and they must be validated:
+                        cfs.indexManager.validateStorageAttachedIndexes(readers);
+                    }
+                    catch (IOException e)
+                    {
+                        throw new RuntimeException(e);
```

Comment:
At this point, I'm sure sure if there's a better alternative than `RuntimeException`...

Reply:
I think `RuntimeException` is fine.

Reply:
Or maybe we want to add our own `CurruptIndexException`, but I don't think we have something particular to do with it.

Reply:
It would be nice if Lucene's `CorruptIndexException` wasn't an `IOException`.

Actually, we could just wrap it in an `UncheckedIOException`. I might do that...

Reply:
`UncheckedIOException` looks like what we need here, +1